src/lib.rs

```diff
@@ -98,6 +104,12 @@ impl fmt::Display for SmolStr {
     }
 }
 
+impl From<String> for SmolStr {
+    fn from(text: String) -> Self {
+        SmolStr(Repr::new_heap(text))
+    }
```

Comment:
Hm, I think if the String is short here, it’s still better to create an inline repr. Let’s change the new arg from str to Into<String> + AsRef<srt>? I think that should be able to reuse allocation if necessary

Reply:
That's a good idea!